web/src/components/users/FirstUser.test.jsx

```diff
@@ -276,3 +286,104 @@ describe("when the user has been modified", () => {
     screen.getByText("ytm");
   });
 });
+
+describe("username suggestions", () => {
+  it("shows suggestions when full name is given and username gets focus", async () => {
+    const { user, dialog } = await openUserForm();
+
+    const fullNameInput = within(dialog).getByLabelText("Full name");
+    await user.type(fullNameInput, "Jane Doe");
+
+    await user.tab();
+
+    const menuItems = screen.getAllByText("Use suggested username");
+    expect(menuItems.length).toBe(4);
+  });
+
+  it("hides suggestions when username loses focus", async () => {
+    const { user, dialog } = await openUserForm();
+
+    const fullNameInput = within(dialog).getByLabelText("Full name");
+    await user.type(fullNameInput, "Jane Doe");
+    
+    await user.tab();
+
+    let menuItems = screen.getAllByText("Use suggested username");
+    expect(menuItems.length).toBe(4);
+
+    await user.tab();
+
+    menuItems = screen.queryAllByText("Use suggested username");
+    expect(menuItems.length).toBe(0);
+  });
+
+  it("does not show suggestions when full name is not given", async () => {
+    const { user, dialog } = await openUserForm();
+
+    const fullNameInput = within(dialog).getByLabelText("Full name");
+    fullNameInput.focus();
+
+    await user.tab();
+
+    const menuItems = screen.queryAllByText("Use suggested username");
+    expect(menuItems.length).toBe(0);
+  });
+
+  it("hides suggestions if user types something", async () => {
+    const { user, dialog } = await openUserForm();
+
+    const fullNameInput = within(dialog).getByLabelText("Full name");
+    await user.type(fullNameInput, "Jane Doe");
+
+    await user.tab();
+
+    // confirming that we have suggestions
+    let menuItems = screen.queryAllByText("Use suggested username");
+    expect(menuItems.length).toBe(4)
+
+    const usernameInput = within(dialog).getByLabelText("Username");
+    // the user now types something
+    await user.type(usernameInput, "John Smith");
+
+    // checking if suggestions are gone
+    menuItems = screen.queryAllByText("Use suggested username");
+    expect(menuItems.length).toBe(0)
+  });
+
+  it("fills username input with chosen suggestion", async () => {
+    const { user, dialog } = await openUserForm();
+
+    const fullNameInput = within(dialog).getByLabelText("Full name");
+    await user.type(fullNameInput, "Jane Doe");
+
+    await user.tab();
+
+    const menuItem = screen.getByText('janed');
+    const usernameInput = within(dialog).getByLabelText("Username");
+    
+    await act(async () => {
+      menuItem.click();
+    });
+
+    expect(usernameInput.value).toBe("janed");
```

Comment:
CSpell complaining here and a few lines above :facepalm: 

You can do either, to add the `/* cspell:disable-next-line */` comment before these lines... or to cheat it by filling the full name input with "User Name" and then clicking on "username" :sweat_smile: :scream: